apps/web/src/app/[locale]/challenge/_components/comments/comment.action.ts

```diff
@@ -13,6 +13,7 @@ interface CommentToCreate {
   text: string;
   rootType: CommentRoot;
   rootId: number;
+  rootChallengeId: number;
```

Comment:
do we need any changes to this file if we can just get the challenge id from the url?

Reply:
Whoops.

Reply:
We need this passed in when `rootId` refers to a `solution id`. `<Comment type="SOLUTION" />`

Are you saying we should grab the challenge id from the URL in this server file?
Because `useParams` is only available to client components.

We'd have to use fetch and grab it from the request instead of using a server function if we want to not pass this explicitly iirc.

Reply:
challengeId is available in the server component via the prop (id)

Reply:
oh wait… hmm okay

let me think about this for a moment

Reply:
isn’t the url 

challenge/{id}/solution/{solutionId}

{id} should be reliable here?

Reply:
i put the comment in the wrong line lol

Reply:
Ah got ya, I was following what you did before for normal comments (Where we save the challenge Id)

Note I did not change the schema at all, just following what I thought was your convention.

Since solutions are tied to challenges, it made sense for me to include the challenge Id as that's how the normal comments worked as well.

Reply:
I guess we could avoid saving it if we read it from the component that handles generating the share URL.

But I felt since we're saving the challenge Id for normal comments, not saving it for other comments will be weird.

I guess you want to retroactively fix sharing for all the comments you already have without messing with the database?

Reply:
solutions already have a relationship with a challenge which is why we don’t need that field to be populated. it’s redundant

Reply:
I think I understand, let me know if I'm on the right track with the latest changes.

https://github.com/typehero/typehero/pull/661/commits/9ff5ed18a0ea142e10348a2ef085d3a8109f241c